### PR TITLE
feat: implement the broadcaster

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Closes https://github.com/celestiaorg/orchestrator-relayer/issues/114

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by evan-forbes — APPROVED** [review]
nice! :+1: :+1:

File: orchestrator/broadcaster.go [hunk 1]
```diff
@@ -3,12 +3,29 @@ package orchestrator
...
+type Broadcaster struct {
+	QgbDHT *p2p.QgbDHT
+}
+
+func NewBroadcaster(qgbDHT *p2p.QgbDHT) *Broadcaster {
+	return &Broadcaster{QgbDHT: qgbDHT}
 }
 
-// Note: broadcaster implementation will be done after defining the P2P interfaces.
+func (b Broadcaster) BroadcastDataCommitmentConfirm(ctx context.Context, nonce uint64, confirm types.DataCommitmentConfirm) error {
```
Comment (evan-forbes) [inline orchestrator/broadcaster.go:19]: minor optional nit on naming, perhaps "provide", or "put" would be better here

File: orchestrator/broadcaster_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBroadcastDataCommitmentConfirm(t *testing.T) {
+	network := qgbtesting.NewDHTNetwork(context.Background(), 4)
+	defer network.Stop()
+
+	// create a test DataCommitmentConfirm
+	expectedConfirm := types.DataCommitmentConfirm{
+		EthAddress: "celes1qktu8009djs6uym9uwj84ead24exkezsaqrmn5",
```
[the remark this answers is not in the document]
Comment (SweeXordious) [reply orchestrator/broadcaster_test.go]: [6dea8bc](https://github.com/celestiaorg/orchestrator-relayer/pull/115/commits/6dea8bcd343e683a2decc96b3bc824634be7768f)
I lost it when I was rebasing, good catch :+1: :+1:

File: orchestrator/broadcaster.go (hunk 1 above)
Comment (SweeXordious) [reply orchestrator/broadcaster.go]: [76e7743](https://github.com/celestiaorg/orchestrator-relayer/pull/115/commits/76e77432321b3f7cd34409673e7fd42a7db2b8c0)

**Pull request merged**